src/pytom_tm/template.py

```diff
@@ -59,32 +59,37 @@ def generate_template_from_map(
         logging.warning(warning_text)
         filter_to_resolution = 2 * output_spacing
 
+    if center:
+        volume_center = np.divide(np.subtract(input_map.shape, 1), 2, dtype=np.float32)
+        # square input to make values positive for center of mass
+        input_center_of_mass = center_of_mass(input_map ** 2)
+        shift = np.subtract(volume_center, input_center_of_mass)
+        input_map = vt.transform(input_map, translation=shift, device='cpu')
+
+        logging.debug(f'center of mass, before was '
+                      f'{np.round(input_center_of_mass, 2)} '
+                      f'and after {np.round(center_of_mass(input_map ** 2), 2)}')
+
     # extend volume to the desired output size before applying convolutions!
     if output_box_size is not None:
-        logging.debug(f'size check {output_box_size} > {(input_map.shape[0] * input_spacing) // output_spacing}')
+        logging.debug(
+            f'size check {output_box_size} > {(input_map.shape[0] * input_spacing) // output_spacing}')
         if output_box_size > (input_map.shape[0] * input_spacing) // output_spacing:
-            pad = int(output_box_size * (output_spacing / input_spacing)) - input_map.shape[0]
+            pad = int(output_box_size * (output_spacing / input_spacing)) - \
+                  input_map.shape[0]
```

Comment:
I prefer parenthesis over backslashes
```suggestion
            pad = (int(output_box_size * (output_spacing / input_spacing)) - 
                   input_map.shape[0])
```